[FrontDoor Management Client] Update WAF and NetworkExperiment SDKs

eschwabe opened the pull request:
- Add exclusions to Web Application Firewall managed rule sets, groups, and rules.
  - Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/7221
- Add create, update, and get operations for Network Experiment profiles and experiments.
  - Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/7370

Commit:
Update FrontDoor management client using latest APIs

Commit:
Add additional checks to WAF policy validation

Commit:
Update package release notes

eschwabe commented:
FYI. @jessicl-ms, @Diego-Perez-Botero, @digavino

Diego-Perez-Botero commented on sdk/frontdoor/Microsoft.Azure.Management.FrontDoor/tests/ScenarioTests/NetworkExperimentTests.cs:
```diff
@@ -0,0 +1,130 @@
+    public class NetworkExperimentTests
+    {
+        [Fact]
+        public void NetworkExperimentCRUDTest()
```
>NetworkExperimentCRUDTest [](start = 20, length = 25)

Thanks for writing this test!! :)

Diego-Perez-Botero reviewed (approved):
:shipit:

Commit:
Move network experiment session recording to expected location

isra-fel commented:
Hi @eschwabe , could you
- Add a link to the swagger spec review PR(s) in the description
- Include the `.txt` file generated when you run `generate.ps1`[(details)](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#sdk-generation-from-updated-spec)
- Use version `2.0.0` since there are breaking changes
- Update `AssemblyInfo.cs` as well

Thanks

Diego-Perez-Botero commented:
@eschwabe - Swagger spec review PR for NetworkExperiment is: https://github.com/Azure/azure-rest-api-specs/pull/7370

Commit:
Update package version and include generated code log

Commit:
Update AzureApiTags

eschwabe commented:
@isra-fel Addressed your comments. Could you please review the latest iteration? Thanks!

isra-fel commented:
> @isra-fel Addressed your comments. Could you please review the latest iteration? Thanks!

The metadata txt was not in the right place, should be `eng/mgmt/mgmtmetadata/`.
Please follow [this guide](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#updating-build-tools) to update your build tools then regenreate.

Other changes are good to me. Thanks.

Commit:
Merge branch 'master' of https://github.com/Azure/azure-sdk-for-net into dev-frontdoor-waf-2019-10-01

Commit:
Fix generated file locations using latest build tools

eschwabe commented:
> > @isra-fel Addressed your comments. Could you please review the latest iteration? Thanks!
> 
> The metadata txt was not in the right place, should be `eng/mgmt/mgmtmetadata/`.
> Please follow [this guide](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#updating-build-tools) to update your build tools then regenreate.
> 
> Other changes are good to me. Thanks.

@isra-fel I had an old version of the powershell module installed. The generate code should be fixed now. Thanks!
